crates/datasources/src/mongodb/mod.rs

```diff
@@ -228,11 +239,13 @@ impl MongoDbTableAccessor {
             .client
             .database(&self.info.database)
             .collection(&self.info.collection);
-        let sampler = TableSampler::new(collection);
+        let sampler = TableSampler::new(&collection);
 
-        let schema = sampler.infer_schema_from_sample().await?;
+        let exact_count = collection.count_documents(None, None).await?;
```

Comment:
This does a table scan. It's kind of odd to do a table scan for the purpose of inferring the schema for the purpose of then pushing anything down. 

If we're going to do a table scan here (which I think we should avoid, the the chance that the metadata scan is off is quite low, and certainly much lower than the schema inference missing a large swath of data.